roles/rsyslog/tasks/outputs/elasticsearch/main.yml

```diff
@@ -4,53 +4,77 @@
     - name: Copy local Elasticsearch ca_certs to /etc/rsyslog.d directory in Rsyslog
       copy:
         src: '{{ item.ca_cert_src }}'
-        dest: '{{ rsyslog_config_dir }}'
+        dest: '{{ item.ca_cert | d(rsyslog_config_dir) }}'
       with_items:
         - '{{ rsyslog_output_elasticsearch }}'
-      when:
-        - item.ca_cert_src is defined
-        - item.ca_cert_src | length > 0
+      when: item.ca_cert_src | d()
 
     - name: Copy local Elasticsearch certs to /etc/rsyslog.d directory in Rsyslog
       copy:
         src: '{{ item.cert_src }}'
-        dest: '{{ rsyslog_config_dir }}'
+        dest: '{{ item.cert | d(rsyslog_config_dir) }}'
       with_items:
         - '{{ rsyslog_output_elasticsearch }}'
-      when:
-        - item.cert_src is defined
-        - item.cert_src | length > 0
+      when: item.cert_src | d()
 
     - name: Copy local Elasticsearch keys to /etc/rsyslog.d directory in Rsyslog
       copy:
         src: '{{ item.key_src }}'
-        dest: '{{ rsyslog_config_dir }}'
+        dest: '{{ item.key | d(rsyslog_config_dir) }}'
+      with_items:
+        - '{{ rsyslog_output_elasticsearch }}'
+      when: item.key_src | d()
+
+    - name: Check certs - triplets are not given
+      fail:
+        msg: "Error: you must specify all 3 of ca_cert, cert, key, or all 3 of ca_cert_src, cert_src, key_src in the elasticsearch output named {{ item.name }}"
       with_items:
         - '{{ rsyslog_output_elasticsearch }}'
       when:
-        - item.key_src is defined
-        - item.key_src | length > 0
+        - item.use_cert | d(true)
+        - not ((item.ca_cert | d() and item.cert | d() and item.key | d()) or
+               (item.ca_cert_src | d() and item.cert_src | d() and item.key_src | d()))
 
     - name: Update ca_cert path to /etc/rsyslog.d directory in Rsyslog
       set_fact:
         __rsyslog_elasticsearch_temp: >-
           {{ __rsyslog_elasticsearch_temp | d([]) }} +
-          {{ [ item | combine( { 'ca_cert': item.ca_cert_src | basename ,
-          'cert': item.cert_src | basename, 'key': item.key_src | basename } ) ] }}
+          {{ [ item |
+               combine( { 'ca_cert': item.ca_cert | d(rsyslog_config_dir + '/' + item.ca_cert_src | basename),
+                          'cert': item.cert | d(rsyslog_config_dir + '/' + item.cert_src | basename),
+                          'key': item.key | d(rsyslog_config_dir + '/' + item.key_src | basename) } )
+          ] }}
+      with_items:
+        - '{{ rsyslog_output_elasticsearch }}'
+      when: item.ca_cert_src | d() or item.cert_src | d() or item.key_src | d()
+
+    - name: Check certs - use_cert is true, but no key/certs data are provided
```

Comment:
What is the difference between this check task and the check task `name: Check certs - triplets are not given` ?

Reply:
You are right. There's no difference... :p

`triplets are not given` used to be for this checking.
> input: The user specifies not enough cert/key files
> output: "Error: you must specify all 3 of ca_cert, cert, key, or all 6 of ca_cert, cert, key, ca_cert_src, cert_src, key_src"

But I had to modify it to the current condition to support the following use case. Then, the 2 became identical... I'll delete the second one.
NOTE: allowing 3 of ca_cert_src, cert_src, key_src are specified, but not other 3.
In this case, the key/certs from the local host are copied to rsyslog_config_dir/{{ca_cert_drc|basename}}, etc.